.github/workflows/main.yml

```
@@ -0,0 +1,71 @@
+      run: |
+        source .venv/bin/activate
+        ruff format --check
+
+  publish:
+    needs: lint
+    # Only run on push to the main branch
+    if: github.ref == 'refs/heads/main' 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
```

actionlint

property "python-version" is not defined in object type {}

(expression)